optimize/dataio.py

```diff
@@ -30,21 +30,33 @@ def __init__(self, filename, swap_xz=True):
             tracks['z_end'] = x_end
             tracks['z'] = x
 
+        tracks = tracks[200:250]
         self.track_fields = tracks.dtype.names
-        self.tracks = torch_from_structured(tracks)
-
-        # self.index = {}
-        # all_events = np.unique(self.tracks['eventID'])
-        # for ev in all_events:
-        #     track_set = np.unique(self.tracks[self.tracks['eventID'] == ev]['trackID'])
-        #     self.index[ev] = track_set
-
+        #self.tracks = torch_from_structured(tracks)
+
+        # flat index for eventID and trackID
+        self.index = []
+        num_tracks = 0
+        all_tracks = []
+        all_events = np.unique(tracks['eventID'])
+        for ev in all_events:
+            track_set = np.unique(tracks[tracks['eventID'] == ev]['trackID'])
+            for trk in track_set:
+                # basic track selection
+                trk_msk = (tracks['eventID'] == ev) & (tracks['trackID'] == trk)
+                if max(tracks[trk_msk]['z']) - min(tracks[trk_msk]['z']) > 3:
```

Comment:
Should we make this cut value configurable?

Reply:
see commit 6ac3d72d8a4caf62ff1a5dd1d5d31353a7e57938